compiler/noirc_frontend/src/elaborator/traits.rs

```diff
@@ -215,6 +217,36 @@ impl<'context> Elaborator<'context> {
         // Don't check the scope tree for unused variables, they can't be used in a declaration anyway.
         self.generics.truncate(old_generic_count);
     }
+
+    pub fn resolve_trait_impl_generics(
+        &mut self,
+        trait_impl: &UnresolvedTraitImpl,
+        trait_id: TraitId,
+    ) -> Option<Vec<Type>> {
+        let trait_def = self.interner.get_trait(trait_id);
+        let resolved_generics = trait_def.generics.clone();
+        if resolved_generics.len() != trait_impl.trait_generics.len() {
+            self.push_err(CompilationError::TypeError(TypeCheckError::GenericCountMismatch {
+                item: trait_def.name.to_string(),
+                expected: resolved_generics.len(),
+                found: trait_impl.trait_generics.len(),
+                span: trait_impl.trait_path.span(),
+            }));
+
+            return None;
+        }
+
+        let trait_generics = trait_impl.trait_generics.iter();
+        let resolved_generics = resolved_generics.iter();
+        Some(
+            trait_generics
+                .zip(resolved_generics)
+                .map(|(generic, resolved_generic)| {
+                    self.resolve_type_inner(generic.clone(), &resolved_generic.kind)
+                })
+                .collect(),
+        )
```

Comment:
```suggestion
        let generics = trait_impl.trait_generics.iter().zip(resolved_generics.iter());
        let mapped = generics.map(|(generic, resolved_generic)| {
            self.resolve_type_inner(generic.clone(), &resolved_generic.kind)
        });
        Some(mapped.collect())
```
Sorry, the `Some` on its own line was really bothering me